File: src/main/java/com/heybys/oddments/base/jpa/MoneyConverter.java
package com.heybys.oddments.base.jpa;

import java.math.BigDecimal;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

import com.heybys.oddments.fooddelivery.domain.generic.Money;

@Converter(autoApply = true)
public class MoneyConverter implements AttributeConverter<Money, BigDecimal> {
    @Override
    public BigDecimal convertToDatabaseColumn(Money money) {
        return money.getAmount();
    }

    @Override
    public Money convertToEntityAttribute(BigDecimal amount) {
        return new Money(amount);
    }
}

File: src/main/kotlin/com/heybys/oddments/base/jpa/MoneyConverter.kt
package com.heybys.oddments.base.jpa

import com.heybys.oddments.fooddelivery.domain.generic.Money
import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter
import java.math.BigDecimal

@Converter(autoApply = true)
class MoneyConverter : AttributeConverter<Money, BigDecimal> {

    override fun convertToDatabaseColumn(money: Money?): BigDecimal? {
        return money?.amount
    }

    override fun convertToEntityAttribute(amount: BigDecimal?): Money? {
        return amount?.let { Money(it) }
    }
}
